### PR TITLE
Fix cls transformers inference

### PR DESCRIPTION
**Pull request by sovrasov**
# What does this PR do?

Softmax node now is searched in the output nodes only

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sovrasov** [comment]
@Wovchena are you ok with this approach? If yes, I'll try to implement it in cpp as well

File: model_api/python/openvino/model_api/models/classification.py [hunk 1]
```diff
@@ -243,9 +243,13 @@ def get_multiclass_predictions(self, outputs):
...
     nodes = inference_adapter.model.get_ops()
     softmaxNode = None
+    all_output_names = []
+    for output in inference_adapter.model.outputs:
+        all_output_names.extend(output.names)
     for op in nodes:
         if "Softmax" == op.get_type_name():
-            softmaxNode = op
+            if op.get_friendly_name() in all_output_names:
+                softmaxNode = op
```
Comment (Wovchena) [inline model_api/python/openvino/model_api/models/classification.py:252]: `ov::Model` has `const ov::ResultVector& get_results()` where `using ResultVector = std::vector<std::shared_ptr<ov::op::v0::Result>>;` and `Result` is a subclass of `Op` which is a subclass of `Node`.

Can you try replacing `nodes = inference_adapter.model.get_ops()` with `inference_adapter.model.get_results()`

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: model_api/python/openvino/model_api/models/classification.py (hunk 1 above)
Comment (Wovchena) [reply model_api/python/openvino/model_api/models/classification.py:253]: Well, I guess we can consider there is no other way. So I'm ok with this approach. Just make `all_output_names` to be a set, not list

Comment (sovrasov) [reply model_api/python/openvino/model_api/models/classification.py]: I implemented another version of workaround, @Wovchena please have a look

File: tests/python/accuracy/public_scope.json [hunk 2]
```diff
@@ -292,7 +292,7 @@
             {
                 "image": "coco128/images/train2017/000000000074.jpg",
                 "reference": [
-                    "anomaly_map min:0.5944074988365173 max:1.0;pred_score:1.0;pred_label:Anomalous;pred_mask min:1 max:1;"
+                    "anomaly_map min:0.5944 max:1.0000;pred_score:1.0;pred_label:Anomalous;pred_mask min:1.0000 max:1.0000;"
```
Comment (Wovchena) [inline tests/python/accuracy/public_scope.json:295]: I expected this to happen here: https://github.com/openvinotoolkit/model_api/pull/98#discussion_r1277247488. But the PR seems to be stuck. Can you just change the precision to 3 digits, not 4, to stay consistent with the rest?

Comment (sovrasov) [reply tests/python/accuracy/public_scope.json]: Done

**Review by Wovchena — APPROVED** [review]

**Pull request merged**